File: thirdscreen/src/main/java/com/jaynewstrom/screenswitchersample/thirdscreen/ThirdScreenDialogFactory.java
package com.jaynewstrom.screenswitchersample.thirdscreen;

import android.app.Dialog;
import android.content.Context;

import com.jnewstrom.screenswitcher.dialoghub.ScopedDialogFactory;

final class ThirdScreenDialogFactory extends ScopedDialogFactory<ThirdComponent> {
    ThirdScreenDialogFactory(Context scopedContext) {
        super(scopedContext);
    }

    @Override public Dialog createDialog(Context context, ThirdComponent component) {
        return new ThirdScreenDialog(context, component);
    }
}

File: sample-screen-third/src/main/java/com/jaynewstrom/screenswitchersample/third/ThirdScreenDialogFactory.kt
package com.jaynewstrom.screenswitchersample.third

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import com.jaynewstrom.concrete.Concrete
import com.jaynewstrom.screenswitcher.dialogmanager.DialogFactory
import javax.inject.Inject
import javax.inject.Named

internal class ThirdScreenDialogFactory : DialogFactory {
    override fun createDialog(context: Context): Dialog {
        return ThirdScreenDialog(context)
    }
}

internal class ThirdScreenDialog(context: Context) : AlertDialog(context) {
    @Inject @field:Named("dialogMessage") lateinit var message: String

    init {
        val component = Concrete.getComponent<ThirdComponent>(context)
        component.inject(this)
        setTitle("Third Dialog")
        setMessage(message)
        setButton(AlertDialog.BUTTON_NEGATIVE, "Done") { _, _ -> }
    }
}
